test/types/range/enum/castEnumRange4a.good

```diff
@@ -1,6 +1,5 @@
 castEnumRange4a.chpl:3: warning: Casts between ranges involving 'enum' indices are currently unstable (see issue #22406); consider performing the conversion manually
 castEnumRange4a.chpl:5: warning: Casts between ranges involving 'enum' indices are currently unstable (see issue #22406); consider performing the conversion manually
-castEnumRange4a.chpl:7: warning: Casts between ranges involving 'enum' indices are currently unstable (see issue #22406); consider performing the conversion manually
```

Comment:
why does this no longer warn? should it?

Reply:
This is due to a bug in how our compiler generates warnings, which got triggered by factoring out the warnings into a helper function, namely `checkEnumIdx`.

Until that bug is fixed, we have the tradeoff between 4x code duplication vs. potentially-missing unstable warnings.

Meanwhile I added a .future test, `test/types/range/enum/castEnumRange4b`